WordBreak.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package dictionary;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ *
+ * @author kailabillie
+ */
+public class WordBreak {
+
+    public static void main(String[] args) {
+
+        ArrayList<String> dictionary = new ArrayList<String>();
+        dictionary.add("pear");
+        dictionary.add("salmon");
+        dictionary.add("foot");
+        dictionary.add("prints");
+        dictionary.add("footprints");
+        dictionary.add("leave");
+        dictionary.add("you");
+        dictionary.add("sun");
+        dictionary.add("girl");
+        dictionary.add("enjoy");
+        
+        System.out.println(canSegment("youenjoy", dictionary));
+        System.out.println(canSegment("youleavefootprints", dictionary));
+        System.out.println(canSegment("salmonenjoyapples", dictionary));
+
+    }
+
+    public static boolean canSegment(String sentence, ArrayList<String> dictionary) {
+
+        sentence = sentence.toLowerCase();
```

Comment:
consider performing validation on inputs. 
What if the input string is null? Then you statement in line 40 will throw a run time error.